Make "classic" style have effect

mdboom opened the pull request:
Hopefully fixes failures in #5416

tacaswell commented:
@mdboom Did this get folded into something else?

mdboom commented:
No -- I was testing it in conjuction with #5416, but I think we can/should still merge this as a separate PR first.

mdboom commented:
It's just the symlog image comparison test that is failing now.  Not sure why yet.

mdboom commented:
I think this is ready to go in now.  We should probably merge sooner than later to prevent fewer false positive results from creeping in.

WeatherGod commented on lib/matplotlib/mathtext.py:
```diff
@@ -1240,58 +1240,115 @@ def get_underline_thickness(self, font, fontsize, dpi):
+    # in the STIX fonts, so we have to detect this one separately.
+    if (constants is STIXFontConstants and
+        isinstance(state.font_output, StixSansFonts)):
+        return STIXSansFontConstants
```
Tab this over one so that it doesn't line up with the code block.

Sorry, made that comment in the commit rather than the PR.

WeatherGod commented on lib/matplotlib/testing/decorators.py:
```diff
@@ -146,11 +148,8 @@ def setup_class(cls):
-        # last prior to running the image test.
-        CleanupTest.setup_class()
         cls.original_settings = cls._initial_settings
+        matplotlib.tests.set_font_settings_for_testing()
```
I like this solution to this problem. I got so tripped up by this when I was working on the classic mode PR.

WeatherGod commented:
I haven't personally vetted the conversion from dictionaries to classes, but the unit tests don't fail, so I am guessing that is ok. Besides the one little PEP8 thing, I am +1 (and even that, I don't care too much about). I am a bit curious why the pep8 tests didn't catch that.

tacaswell commented:
Pretty sure that is one of the pep8 errors/warnings we have blacklisted.

WeatherGod commented:
Really? That is actually one of the PEP8 rules that I think makes the most
sense. I get tripped up all the time by seeing clause statements lined up
with code blocks because Python is so whitespace-oriented.

On Mon, Nov 16, 2015 at 10:51 AM, Thomas A Caswell <notifications@github.com
> wrote:

> Pretty sure that is one of the pep8 errors/warnings we have blacklisted.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5437#issuecomment-157076127>
> .
>

jenshnielsen commented:
:+1: It looks like some of the symlog images are changed twice. Would it make sense to squash the commits?

mdboom commented:
Sure, I can squash the commits.

Also, let's hold off -- I think I just discovered another case where this isn't working...

mdboom commented:
Ok -- I've fixed one last thing and now I have my massive style changes branch passing (stay tuned).  (The @cleanup decorator wasn't applying the classic style, which soon will affect things like view limits that even non-image-comparison tests care about).

Also addressed @WeatherGod's indentation comment.

Once Travis passes, I'd say this is good to go.

WeatherGod commented on lib/matplotlib/testing/decorators.py:
```diff
@@ -109,18 +110,27 @@ def tearDownClass(cls):
+        return make_cleanup
+    else:
+        result = make_cleanup(style)
+        style = 'classic'
```
isn't this backwards?

WeatherGod replied:
ok, nevermind... I got mixed up

mdboom replied:
No.  The idea here is to keep `@cleanup` without arguments working.  If you pass it no arguments, the first argument should be a function and we want to use classic mode by default.  If you pass a string it specifies a style, and the function will be passed in later (through function chaining).  This is just typical decorator madness, but I can add a comment if you think it would help.

WeatherGod replied:
Perhaps make a comment here explaining that `style` in some cases isn't really a style name?

mdboom replied:
I've added a comment.

WeatherGod commented:
one of the Travis errors seems spurious, but the other one seems to be some sort of odd stix font issue that only happens in py3.4? I restarted the spurious one, though.

mdboom commented:
No, it should be on 2.0.x now -- everything else we're doing ends up sort of relying on this.  I'll go ahead and do that now.

mdboom commented:
Backported to 2.0.x as 10e2af5
